### DISCUSSION TITLE
DevTools integration: Can I use Runtime.consoleAPICalled?

### DISCUSSION THREAD

**Discussion by OllieJones** [MicrosoftEdge/WebView2Feedback#3902] [Q&A]
I have WebView2 in my project, but not WebView.DevToolsProtocol Extension.  

I use this code to get textual JSON messages from the console.*() calls in Javascript in the embedded browser.

```c#
view.CoreWebView2
    .GetDevToolsProtocolEventReceiver("Console.messageAdded")
    .DevToolsProtocolEventReceived += DevToolsMessages.ConsoleMessage;
await view.CoreWebView2.CallDevToolsProtocolMethodAsync("Console.enable", "{}");

view.CoreWebView2
    .GetDevToolsProtocolEventReceiver("Runtime.consoleAPICalled")
    .DevToolsProtocolEventReceived += DevToolsMessages.ConsoleMessage;
```
The first stanza of code is fine. It delivers the deprecated [Console.messageAdded](https://chromedevtools.github.io/devtools-protocol/tot/Console/#type-ConsoleMessage) messages.

But I don't receive the [Runtime.consoleAPICalled](https://chromedevtools.github.io/devtools-protocol/tot/Runtime/#event-consoleAPICalled) messages. And, I can't find a way to do something similar to calling `Console.enable()` in the protocol to get those messages.

Am I missing something? (Probably...) Any advice?

**Comment by schuylergleaves** [comment]
Hey @OllieJones, you would have to [enable the Runtime domain ](https://chromedevtools.github.io/devtools-protocol/tot/Runtime/#method-enable) via
`await view.CoreWebView2.CallDevToolsProtocolMethodAsync("Runtime.enable", "{}");`

It's a little confusing that the Runtime.enable method only mentions the executionContextCreated event but it is required for any event in that domain.

Hope that helps!
